doc/UsersGuide/BuildingRunningTesting/RunSRW.rst

```diff
@@ -629,75 +617,344 @@ The output files (in ``.png`` format) will be located in the ``postprd`` directo
+file will necessarily be included in the workflow.  This is because, in order to avoid unwanted computation,
+only those (meta)tasks in ``verify_det.yaml`` and/or ``verify_ens.yaml`` that operate on field groups
+included in ``VX_FIELD_GROUPS`` will appear in the Rocoto XML, and only those (meta)tasks in ``verify_pre.yaml``
+that operate on obs types associated with one of the field groups in ``VX_FIELD_GROUPS`` will appear in 
+the Rocoto XML.  Thus, for example, setting
 
 .. code-block:: console
 
-   platform:
-      CCPA_OBS_DIR: /path/to/UFS_SRW_data/develop/obs_data/ccpa/proc
-      NOHRSC_OBS_DIR: /path/to/UFS_SRW_data/develop/obs_data/nohrsc/proc
-      MRMS_OBS_DIR: /path/to/UFS_SRW_data/develop/obs_data/mrms/proc
-      NDAS_OBS_DIR: /path/to/UFS_SRW_data/develop/obs_data/ndas/proc
+   VX_FIELD_GROUPS: [ 'APCP', 'REFC', 'RETOP', 'SFC', 'UPA' ]
+
+in ``config.yaml`` and including all three taskgroups ``verify_*.yaml`` in ``rocoto: tasks: taskgroups:`` 
+will add to the Rocoto XML the VX (meta)tasks for all valid field groups except those for accumulated 
+snowfall (``'ASNOW'``) and its associated obs type (:term:`NOHRSC`).  In other words, all the (meta)tasks
+in :numref:`Table %s <VXWorkflowTasksTable>`.  will be included in the Rocoto XML except for those
+associated with the :term:`NOHRSC` obs type and the ``'ASNOW'`` field group.  Users might want to set
+``VX_FIELD_GROUPS`` in this way for example because the forecast experiment they are verifying is for a
+summer period for which ``ASNOW`` is not relevant.
+
+
+Staging Observation Files
+``````````````````````````````````
+The taskgroup in ``verify_pre.yaml`` defines a set of workflow tasks named ``get_obs_*`` --- where the ``*``
+represents any one of the supported obs types :term:`CCPA`, :term:`NOHRSC`, :term:`MRMS`, and :term:`NDAS` ---
+that will first check for the existence of the obs files required for VX at the locations on disk specified
+by the variables ``*_OBS_DIR`` and ``OBS_*_FN_TEMPLATES[1,3,...]`` in the ``verification:`` section of
+``config.yaml``.  The ``*_OBS_DIR``  are the base directories in which the obs files are or should be
+staged, and the ``OBS_*_FN_TEMPLATES[1,3,...]`` are the file name templates (with METplus time strings
+used for templating; see example below).  The ``[1,3,...]`` in ``OBS_*_FN_TEMPLATES[1,3,...]`` refer
+to the second, fourth, etc elements of ``OBS_*_FN_TEMPLATES`` and correspond to the various sets of files
+that the obs type contains.  (The first, third, etc elements, i.e. ``OBS_*_FN_TEMPLATES[0,2,...]``,
+indicate the VX field groups for which the respective sets of obs files are used to verify and should
+not be changed.) Note that ``OBS_*_FN_TEMPLATES[1,3,...]`` may include leading subdirectories and are
+relative to the obs type's ``*_OBS_DIR``.  Thus, the templates for the full paths to the obs files are
+given by
+
+.. code-block:: console
+
+   {*_OBS_DIR}/{OBS_*_FN_TEMPLATES[1,3,...]}
```

Comment:
```suggestion
The taskgroup in ``verify_pre.yaml`` defines a set of workflow tasks named ``get_obs_*``, where the ``*``
represents any one of the supported obs types: :term:`CCPA`, :term:`NOHRSC`, :term:`MRMS`, and :term:`NDAS`. These ``get_obs_*`` tasks 
will first check on disk for the existence of the obs files required for VX using the locations specified
by the variables ``*_OBS_DIR`` and ``OBS_*_FN_TEMPLATES[1,3,...]`` in the ``verification:`` section of
``config.yaml``. The ``*_OBS_DIR``  are the base directories in which the obs files should be
staged, and the ``OBS_*_FN_TEMPLATES[1,3,...]`` are the file name templates (with METplus time strings
used for templating; see example below). Thus, the templates for the full paths to the obs files are
given by:

.. code-block:: console
   
   {*_OBS_DIR}/{OBS_*_FN_TEMPLATES[1,3,...]}

The ``[1,3,...]`` in ``OBS_*_FN_TEMPLATES[1,3,...]`` refer
to the second, fourth, etc. elements of ``OBS_*_FN_TEMPLATES`` and correspond to the various sets of files
that the obs type contains. (The first, third, etc. elements, i.e., ``OBS_*_FN_TEMPLATES[0,2,...]``,
indicate the VX field groups for which the respective sets of obs files are used to verify and should
not be changed.) Note that ``OBS_*_FN_TEMPLATES[1,3,...]`` may include leading subdirectories and are
relative to the obs type's ``*_OBS_DIR``. 
```

Reply:
@gsketefian I'm a little confused by the `OBS_*_FN_TEMPLATES[1,3,...]` explanation. Are the numbers in the list zero-indexed? If so, it would be good to say that. It also seems like odd-numbered integers may refer to file types and even-numbered integers to field groups? If so, it might be good to state that more clearly. I'm happy to help reword, but I wanted to clarify my own understanding.

Reply:
@gspetro-NOAA This is a bit confusing because I wanted to use python dictionaries for the  ``OBS_*_FN_TEMPLATES``, but it turns out that can't be done unless the ex-scripts for the various vx tasks are converted to python.  (I can explain why that is if you're interested, but it's beside the point.)  I didn't want to make the bash-to-python conversion of the ex-scripts in this PR, so I had to make ``OBS_*_FN_TEMPLATES`` just python lists.  Python lists are 0-indexed, that's why the first element has index 0, etc.  As you pointed out, the contents of ``OBS_*_FN_TEMPLATES`` come in pairs, where the first element of the pair (with index 0, 2, ...) is the field group(s) that the file type is used to verify, while the second element (with index 1, 3, ...) is the METplus template for that file type.

When the ex-scripts are converted to python, we can convert the ``OBS_*_FN_TEMPLATES`` into dictionaries, where the first element of the each pair would be a key and the second element would be its value.  So instead of having this in ``config_defaults.yaml``:
```
  OBS_CCPA_FN_TEMPLATES: [ 'APCP',
                           '{%- set obs_avail_intvl_hrs = "%02d" % CCPA_OBS_AVAIL_INTVL_HRS %}
                            {{- "{valid?fmt=%Y%m%d}/ccpa.t{valid?fmt=%H}z." ~ obs_avail_intvl_hrs ~ "h.hrap.conus.gb2" }}' ]
  OBS_NOHRSC_FN_TEMPLATES: [ 'ASNOW',
                             '{%- set obs_avail_intvl_hrs = "%d" % NOHRSC_OBS_AVAIL_INTVL_HRS %}
                              {{- "sfav2_CONUS_" ~ obs_avail_intvl_hrs ~ "h_{valid?fmt=%Y%m%d%H}_grid184.grb2" }}' ]
  OBS_MRMS_FN_TEMPLATES: [ 'REFC', '{valid?fmt=%Y%m%d}/MergedReflectivityQCComposite_00.50_{valid?fmt=%Y%m%d}-{valid?fmt=%H%M%S}.grib2',
                           'RETOP', '{valid?fmt=%Y%m%d}/EchoTop_18_00.50_{valid?fmt=%Y%m%d}-{valid?fmt=%H%M%S}.grib2' ]
  OBS_NDAS_FN_TEMPLATES: [ 'SFCandUPA', 'prepbufr.ndas.{valid?fmt=%Y%m%d%H}' ]
```
we would instead have
```
  OBS_CCPA_FN_TEMPLATES: { 'APCP':
                           '{%- set obs_avail_intvl_hrs = "%02d" % CCPA_OBS_AVAIL_INTVL_HRS %}
                            {{- "{valid?fmt=%Y%m%d}/ccpa.t{valid?fmt=%H}z." ~ obs_avail_intvl_hrs ~ "h.hrap.conus.gb2" }}' }
  OBS_NOHRSC_FN_TEMPLATES: { 'ASNOW':
                             '{%- set obs_avail_intvl_hrs = "%d" % NOHRSC_OBS_AVAIL_INTVL_HRS %}
                              {{- "sfav2_CONUS_" ~ obs_avail_intvl_hrs ~ "h_{valid?fmt=%Y%m%d%H}_grid184.grb2" }}' }
  OBS_MRMS_FN_TEMPLATES: { 'REFC': '{valid?fmt=%Y%m%d}/MergedReflectivityQCComposite_00.50_{valid?fmt=%Y%m%d}-{valid?fmt=%H%M%S}.grib2',
                           'RETOP': '{valid?fmt=%Y%m%d}/EchoTop_18_00.50_{valid?fmt=%Y%m%d}-{valid?fmt=%H%M%S}.grib2' }
  OBS_NDAS_FN_TEMPLATES: { 'SFCandUPA': 'prepbufr.ndas.{valid?fmt=%Y%m%d%H}' }
```
 Feel free to reword my explanation if you like, or I can do it if you prefer.  The only change I'd suggest in your current rewording is to change "in which the obs should be staged" to "in which the obs **are or** should be staged" since these variables can either point to already-staged files (the "are") or tell the `get_obs` tasks where to put the files (the "should be").  You don't have to word it exactly this way, but I do want to get that point across somehow.  Thanks!
 .

Reply:
How's this rewrite?

Reply:
@gsketefian -

It looks like this is the last unresolved conversation in the PR.  If @gspetro-NOAA's rewrite is okay, please commit the suggested change.  Once the Jenkins tests finish on Derecho, I will be able to merge this PR at that time.  Thank you very much!